src/parquet/parquet.h

```diff
@@ -117,6 +117,22 @@ class InMemoryInputStream : public InputStream {
   int64_t offset_;
 };
 
+// A wrapper for InMemoryInputStream to manage the memory.
+class ScopedInMemoryInputStream : public InputStream {
+ public:
+  ScopedInMemoryInputStream(const int64_t len);
```

Comment:
This is the length of the buffer allocated by ScopedInMemoryInputStream.
@wesm Since you have incorporate this PR into #23, can you please add a comment to #23?